File: src/main/java/uk/gov/justice/digital/hmpps/oauth2server/security/AuthSource.java
package uk.gov.justice.digital.hmpps.oauth2server.security;

import com.fasterxml.jackson.annotation.JsonValue;

import java.util.Optional;

public enum AuthSource {
    nomis, auth, delius, azuread, none;

    @JsonValue
    public String getSource() {
        return name();
    }

    public static AuthSource fromNullableString(final String source) {
        return Optional.ofNullable(source).map(s -> valueOf(source.toLowerCase())).orElse(none);
    }
}

File: src/main/java/uk/gov/justice/digital/hmpps/oauth2server/security/AuthSource.kt
package uk.gov.justice.digital.hmpps.oauth2server.security

import com.fasterxml.jackson.annotation.JsonValue

enum class AuthSource {
  auth, azuread, delius, nomis, none;

  companion object {
    @JvmStatic
    fun fromNullableString(source: String?): AuthSource {
      return source?.let { valueOf(source.toLowerCase() ) } ?: none
    }
  }

  @JsonValue
  fun getSource(): String {
    return name
  }
}